### DISCUSSION TITLE
apollo是否支持超级管理员权限api

### DISCUSSION THREAD

**Discussion by codwu** [apolloconfig/apollo#3988] [Q&A]
我们希望能通过api的方式操作apollo，通过开放平台的方式，似乎只能在操作单个应用，创建应用、授权等还是得在页面上操作。apollo有没有超级管理员权限的api呢？例如可以通过api创建应用、授权等等

**Comment by nobodyiam** [comment]
目前尚不支持，主要考虑的因素点是这个权限太大，通过 api 访问的话未必能受控，存在潜在风险，不过大家可以讨论下

**Comment by zouyx** [comment]
好像不是太合适，毕竟大部分有安全部的公司。都可能过不去安全审计。

**Comment by codwu** [comment]
那就必须用apollo的portal了？我们想自研一个portal的话该怎么做呢？

**Reply by nobodyiam** [reply]
自研 portal 是作为一个第三方的补充还是完全替代 apollo-portal？如果是第三方的补充，比如解决部分场景的配置管理问题如数据库，那么 open api 目前的权限应该是够的，如果是要完全替代 apollo-portal，那可以建议直接基于 apollo-portal 定制？

**Comment by codwu** [comment]
很多开源产品其实都有高权限的API的，至于安全性，可以从认证和授权方式的角度上来提高

**Comment by Anilople** [comment]
可否放开创建应用的权限

存在1个token，可以用来创建应用，但是无法删除应用，无法改应用配置。portal的配置信息做对应的开关是否打开这个feature，默认关闭，想使用的公司可以自行打开。

场景：

有些金融公司测试和生产的portal是分开的，上线需要使用word文档写生产操作手册，由于第一次上生产时需要创建应用，手册会写得比较麻烦，

此时如果可以让shell脚本创建应用，会很方便。如果直接在portal的db执行sql感觉不太好；用portal管理员用户登录后的cookie然后调用相关的api也不太安全；

**Reply by nobodyiam** [reply]
如果只是创建应用的权限，那应该是可控的，不过创建完后还涉及了对应用的管理，比如更新配置、发布配置等，这部分权限在测试和生产的 portal 分开场景下是怎么做的？管理员单独去应用的 namespace 中配置？
这部分如果是要 openapi 同步的话，同样也会涉及权限的管控，比如是有一个超级管理员权限，还是每个 namespace 单独管理权限。
